File: AppConHilosJavaEjercicios/Ejercicio1/Programa1.java
package Ejercicio1;

public class Programa1 {

    public static void main (String[] args) throws InterruptedException{
        Hilo hilo1 = new Hilo("Hilo1", 1, 100);
        Hilo hilo2 = new Hilo("Hilo2", 100, 1);

        Thread h1 = new Thread(hilo1);
        Thread h2 = new Thread(hilo2);

        h1.start();
        h2.start();
    }
    
}

File: AppConHilosPython/programa1.py
import threading
import os

# Función para el hilo 1
def task1():
    print("Tarea 1 asignada al thread: {}".format(threading.current_thread().name))
    print("ID del proceso corriendo la tarea 1: {}".format(os.getpid()))

# Función para el hilo 2
def task2():
    print("Tarea 2 asignada al thread: {}".format(threading.current_thread().name))
    print("ID del proceso corriendo la tarea 2: {}".format(os.getpid()))

# Aunque ya sabemos que este es el programa principal,
# ejemplificamos el uso de la variable __name__
if __name__ == "__main__":
    print("ID del proceso corriendo el programa main: {}".format(os.getpid()))
    print("Nombre del Main thread: {}".format(threading.current_thread().name))

# Creamos los hilos
t1 = threading.Thread(target=task1, name="Hilo1")
t2 = threading.Thread(target=task2, name="Hilo2")

# Iniciamos los hilos
t1.start()
t2.start()

print("Programa terminado!")